src/formatDate.js

```diff
@@ -8,7 +8,37 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.slice(-1)[0];
+  const toSeparator = toFormat.slice(-1)[0];
+  const dateElements = date.split(fromSeparator);
+
+  const dateMap = fromFormat.reduce((acc, format, index) => {
+    acc[format] = dateElements[index];
+
+    return acc;
+  }, {});
+
+  const convertYear = (year, format) => {
+    if (format === 'YYYY') {
+      return year.length === 2 ? (year < 30 ? '20' + year : '19' + year) : year;
+    }
+
+    if (format === 'YY') {
+      return year.slice(-2);
+    }
+
+    return year;
+  };
```

Comment:
The `convertYear` function works well for the given task. However, it's always good practice to explicitly parse strings to numbers before performing numeric comparisons. In JavaScript, '8' < '30' is false because it compares strings in lexicographical order. Use `parseInt` to convert the `year` string to a number before the comparison.